File: C Sharp/Blink/Sample/CallBack.cs
using Net.Qiujuer.Blink.Box;
using Net.Qiujuer.Blink.Core;
using Net.Qiujuer.Blink.Listener;
using Net.Qiujuer.Blink.tool;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sample
{
    class CallBack : ReceiveListener, Executor
    {
        public void execute(Runnable command)
        {
            command.run();
        }

        public void OnReceiveStart(int type, long id)
        {
            Console.WriteLine("Receive->start:" + type + " " + id);
        }

        public void OnReceiveProgress(int type, long id, int total, int cur)
        {
            Console.WriteLine("Receive->progress:" + type + " " + id
                         + " " + total + " " + cur);
        }

        public void OnReceiveEnd(Net.Qiujuer.Blink.Core.ReceivePacket paket)
        {
            if (paket.GetType() == BlinkPacket.Type.STRING)
                Console.WriteLine("Receive->end: String:"
                        + paket.GetId() + " " + paket.GetLength() + " :"
                        + ((StringReceivePacket)paket).GetEntity());
            else if (paket.GetType() == BlinkPacket.Type.BYTES)
                Console.WriteLine("Receive->end: Bytes:"
                        + paket.GetId() + " " + paket.GetLength() + " :"
                        + ((ByteReceivePacket)paket).GetEntity());
            else
                Console.WriteLine("Receive->end: File:"
                        + paket.GetId()
                        + " "
                        + paket.GetLength()
                        + " :"
                        + ((FileInfo)(((FileReceivePacket)paket).GetEntity()))
                        .FullName + " " + paket.GetHashCode());
        }
    }
}

File: Java/Blink/sample/src/net/qiujuer/sample/blink/CallBack.java
/*
 * Copyright (C) 2014 Qiujuer <qiujuer@live.cn>
 * WebSite http://www.qiujuer.net
 * Created 04/19/2015
 * Changed 04/19/2015
 * Version 1.0.0
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.qiujuer.sample.blink;

import net.qiujuer.blink.box.ByteReceivePacket;
import net.qiujuer.blink.box.FileReceivePacket;
import net.qiujuer.blink.box.StringReceivePacket;
import net.qiujuer.blink.core.Connector;
import net.qiujuer.blink.core.PacketType;
import net.qiujuer.blink.core.ReceivePacket;
import net.qiujuer.blink.core.listener.ConnectListener;
import net.qiujuer.blink.core.listener.ReceiveListener;

import java.util.Arrays;

/**
 * Call Back status
 */
public class CallBack implements ReceiveListener, ConnectListener {

    @Override
    public void onReceiveStart(Connector connector, ReceivePacket packet) {
        System.out.println("Receive->start:" + packet.getId() + " " + packet.getPacketType());
    }

    @Override
    public void onReceiveProgress(Connector connector, ReceivePacket packet, float progress) {
        System.out.println("Receive->progress:" + progress);
    }

    @Override
    public void onReceiveCompleted(Connector connector, ReceivePacket packet) {
        if (packet.getPacketType() == PacketType.STRING)
            System.out.println("Receive->end: String:"
                    + packet.getId() + " " + packet.getLength() + " :"
                    + ((StringReceivePacket) packet).getEntity());
        else if (packet.getPacketType() == PacketType.BYTES)
            System.out.println("Receive->end: Bytes:"
                    + packet.getId() + " " + packet.getLength() + " :"
                    + Arrays.toString(((ByteReceivePacket) packet).getEntity()));
        else
            System.out.println("Receive->end: File:"
                    + packet.getId()
                    + " "
                    + packet.getLength()
                    + " :"
                    + ((FileReceivePacket) packet).getEntity()
                    .getPath() + " " + packet.getHash());
    }

    @Override
    public void onConnectClosed(Connector connector) {
        System.out.println("A BlinkDisconnect.");
    }
}
